File: backend/app/schemas/workspace.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WorkspaceCreate(BaseModel):
    name: str
    description: str | None = None


class WorkspaceUpdate(BaseModel):
    name: str | None = None
    description: str | None = None


class WorkspaceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str
    description: str | None
    created_at: datetime
    updated_at: datetime


class WorkspaceMemberResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    workspace_id: int
    user_id: int
    role: str
    created_at: datetime


class AddWorkspaceMemberRequest(BaseModel):
    email: str
    role: str = "viewer"


class UpdateWorkspaceMemberRoleRequest(BaseModel):
    role: str

File: frontend/src/types/workspace.ts
export type WorkspaceRole = 'owner' | 'admin' | 'editor' | 'viewer';

export type AssignableWorkspaceRole = 'admin' | 'editor' | 'viewer';

export interface Workspace {
  id: number;
  name: string;
  slug: string;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface WorkspaceMember {
  id: number;
  workspace_id: number;
  user_id: number;
  username: string;
  email: string;
  role: WorkspaceRole;
  created_at: string;
}

export interface CreateWorkspaceBody {
  name: string;
  description?: string;
}

export interface UpdateWorkspaceBody {
  name?: string;
  description?: string | null;
}

export interface AddMemberBody {
  email: string;
  role: AssignableWorkspaceRole;
}

export interface UpdateMemberRoleBody {
  role: AssignableWorkspaceRole;
}

export interface MessageResponse {
  message: string;
}
